### ISSUE TITLE
Plugin using wrong cloudformation-cli-python-lib

### ISSUE THREAD

**Issue by wnkz** [aws-cloudformation/cloudformation-cli-python-plugin#112]
Hi,

When building Python custom resources we are mainly using three libraries:

- https://pypi.org/project/cloudformation-cli/
- https://pypi.org/project/cloudformation-cli-python-plugin/
- https://pypi.org/project/cloudformation-cli-python-lib/

Since the end of development preview for Python, we had python-plugin 2.0.0 and python-lib 2.0.0.

Both packages are located in this repository (https://github.com/aws-cloudformation/cloudformation-cli-python-plugin)

- cloudformation-cli-python-plugin: setup.py
- cloudformation-cli-python-lib: src/setup.py

In order to create a Python resource with cloudformation-cli, we need cloudformation-cli and cloudformation-cli-python-plugin installed. Everything is fine until there.

After running `cfn init` ; we end up with a `requirements.txt` file containing this:

```
cloudformation-cli-python-lib==2.0.0
```

This file will later be used by SAM to package dependencies for the Lambda function.

Now if we try to build our resource with `cfn generate && cfn submit --dry-run` we end up with a `build/` directory containing everything our Lambda function needs.
The problem is that when SAM built the function, it read `requirements.txt` file which is requiring a package called `cloudformation-cli-python-lib` on PyPi (https://pypi.org/project/cloudformation-cli-python-lib/).

So in the end we expect something like this:

https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/d00dd26c9f7a4a1baf3f134a3a12f4daac19d775/src/cloudformation_cli_python_lib/utils.py#L111-L121

But what we actually have in the build directory is like this:

https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/21a7da931208dc43af60d2688ead48d66c7eaa72/src/cloudformation_cli_python_lib/utils.py#L111-L117

----

## TLDR

cloudformation-cli-python-plugin and cloudformation-cli-python-lib are in the same repository but cloudformation-cli-python-lib package was not updated on PyPi with 2.1.0 release and now it is impossible to run tests because SAM uses PyPi to install requirements into the Lambda's build directory.

I think this **critical** because we cannot run tests anymore **even after I rolled back to a previous CLI version** (<=0.1.6) I get the following error when running `cfn test`:

```
ERROR: usage: cfn [options] [file_or_dir] [file_or_dir] [...]
cfn: error: unrecognized arguments: --no-print-logs
  inifile: /private/var/folders/bf/f_z6y6n959193x3csh_pcb0m0000gn/T/pytest_0da_6_7q.ini
  rootdir: /path/to/my/resource

One or more contract tests failed
```

This issue is linked to #109

**Comment by z0ph** [comment]
Up!

Same issue here.

**Comment by wnkz** [comment]
No sign of life from any of the maintainers for 18 days 🤔 No activity either in #109, duplicated PR for #114 and not a single test is passing. Is Python language still maintained for cloudformation-cli or should we just learn Java/Go?

**Comment by jotompki** [comment]
Apologies for the delay. We will look into releasing a new version of the support library and cli plugin to address. This coupled with releasing the cloudformation-cli with [this change](https://github.com/aws-cloudformation/cloudformation-cli/pull/533) should unblock testing via cfn test.

**Comment by wnkz** [comment]
Thanks, looking forward to those changes

**Comment by ammokhov** [comment]
tracking #109 here

**Comment by jotompki** [comment]
Addressed in #120. Latest tagged releases of the plugin and support lib contain the changes necessary. Will release to PyPI shortly.

**Comment by jotompki** [comment]
Released to PyPI. please let me know if you are still encountering issues with the latest lib versions:

https://pypi.org/project/cloudformation-cli-python-lib/2.1.1/

https://pypi.org/project/cloudformation-cli-python-plugin/2.1.1/

**Comment by wnkz** [comment]
@jotompki I'm sorry but it **still** does not work.. I can open another issue if needed but running `cfn test` with newest versions (0.1.10 / 2.1.1 / 2.1.1) gives me:

```
START RequestId: 06295245-25d8-1d02-c0e6-020f1039210a Version: $LATEST
[ERROR] 2020-09-15T12:53:51.219Z        06295245-25d8-1d02-c0e6-020f1039210a    Invalid request
Traceback (most recent call last):
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 142, in _parse_request
    event = HandlerRequest.deserialize(event_data)
  File "/var/task/cloudformation_cli_python_lib/utils.py", line 98, in deserialize
    event = HandlerRequest(**json_data)
TypeError: __init__() missing 1 required positional argument: 'responseEndpoint'
Handler error
Traceback (most recent call last):
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 142, in _parse_request
    event = HandlerRequest.deserialize(event_data)
  File "/var/task/cloudformation_cli_python_lib/utils.py", line 98, in deserialize
    event = HandlerRequest(**json_data)
TypeError: __init__() missing 1 required positional argument: 'responseEndpoint'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 187, in __call__
    sessions, action, callback, event = self._parse_request(event_data)
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 150, in _parse_request
    raise InvalidRequest(f"{e} ({type(e).__name__})") from e
cloudformation_cli_python_lib.exceptions.InvalidRequest: __init__() missing 1 required positional argument: 'responseEndpoint' (TypeError)
END RequestId: 06295245-25d8-1d02-c0e6-020f1039210a
REPORT RequestId: 06295245-25d8-1d02-c0e6-020f1039210a  Init Duration: 2499.98 ms       Duration: 12.14 ms      Billed Duration: 100 ms Memory Size: 128 MB     Max Memory Used: 39 MB
```

This does not even reach my code 😕
